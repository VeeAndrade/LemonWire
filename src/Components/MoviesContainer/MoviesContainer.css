.movies-display-container {
  display: flex;
  flex-flow: column nowrap;
  height: 100vh;
  margin-top: 80px;
  width: 100%;
}

.top-rated-movies {
  /* border: solid 2px red; */
  width: 100%;
}

.bottom-display-section {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
}
