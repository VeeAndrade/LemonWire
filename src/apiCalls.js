export const getUserInfo = user => {
  const options = {
    method: 'POST',
    body: JSON.stringify(user),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  return fetch('https://rancid-tomatillos.herokuapp.com/api/v1/login', options)
    .then(response => {
    if (!response.ok) {
      throw Error('200 status code not found: getUserInfo throw error')
    }
    return response.json()
  })
}

export const getUserRatings = userId => {
  return fetch(`https://rancid-tomatillos.herokuapp.com/api/v1/users/${userId}/ratings`)
  .then(response => {
    if (!response.ok) {
      throw Error('200 status code not found: getUserRatings throw error')
    }
    return response.json()
  })
}