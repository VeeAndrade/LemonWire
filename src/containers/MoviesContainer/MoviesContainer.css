.movies-display-container {
  display: flex;
  flex-flow: column nowrap;
  margin-top: 80px;
  width: 100%;
}

.movies-heading {
  background-color: #E6D532;
  color: #292B33;
  width: 20%;
  text-align: center;
  height: 2rem;
  margin-top: 1em;
  margin-bottom: 1em;
  border-radius: 0 35px 35px 0;
  display: flex;
  align-items: center;
  justify-content: center;
}

.top-rated-movies {
  width: 100%;
  display: flex;
  align-self: center;
  flex-direction: row;
  flex-wrap: nowrap;
  height: 25rem;
  overflow-x: scroll;
  background-image: linear-gradient(rgba(24, 24, 24, 0.838), rgba(12, 12, 12, 0.831)),url(../../Assets/black-and-gold.jpg);
  background-size: cover;
  background-position: top;
  background-repeat: no-repeat;
}

.bottom-display-section {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;
}
