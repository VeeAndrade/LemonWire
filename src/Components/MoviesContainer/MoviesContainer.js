import React from 'react';
import './MoviesContainer.css';
import MoviePreview from '../MoviePreview/MoviePreview';

const MoviesContainer = ({ movies }) => {
  let allMovies =  movies.map(movie => {
    return <MoviePreview key={movie.id} movie={movie}/>
  })

  let filteredMovies =  movies.filter(film => {
    return film.average_rating > 5
  })

  let topMovies = filteredMovies.map(movie => {
    return <MoviePreview key = {movie.id} movie = {movie}/>
  })

  return (
    <section className='movies-display-container'>
      <section className='top-rated-movies'>

      </section>
      <section className='bottom-display-section'>
        {allMovies}
      </section>
    </section>
  )
}

export default MoviesContainer;
