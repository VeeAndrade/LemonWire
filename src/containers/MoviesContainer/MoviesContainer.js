import React, { Component } from 'react';
import './MoviesContainer.css';
import MoviePreview from '../../Components/MoviePreview/MoviePreview';
import { connect } from 'react-redux';
import { getMovies } from '../../actions'
import { updateSelectedMovie } from '../../actions';


class MoviesContainer extends Component {

  componentDidMount() {
    fetch('https://rancid-tomatillos.herokuapp.com/api/v1/movies')
      .then(response => response.json())
      .then(movies => {
        this.props.addMoviesToStore(movies.movies)})
      .catch(error => console.log(error))
  }

  formatDate(releaseDate) {
     const monthNames = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
    let date = releaseDate.split('-')
    return `${monthNames[date[1] - 1]} ${date[2]}, ${date[0]}`;
  }

  findUserRating = movie => {
    return this.props.ratings.find(rating => rating.movie_id === movie) ? this.props.ratings.find(rating => rating.movie_id === movie).rating : 0;

  }

  render() {
    
    let allMovies = this.props.movies.map(movie => {
      movie.user_rating = this.props.ratings ? this.findUserRating(movie.id) : 0;
      return <MoviePreview saveSelectedMovieToStore={this.saveSelectedMovieToStore} key={movie.id} date={this.formatDate(movie.release_date)} movie={movie}/>})

    let sortedMovies = this.props.movies.sort((a, b) => a.average_rating - b.average_rating)
    let topMovies = sortedMovies.map(movie => <MoviePreview key = {movie.id} movie = {movie}/>)

    return (
      !this.props.movies ? <p>loading</p> : (
      <section className='movies-display-container'>
        <section className='movies-heading'>
          <h3>Highy Rated Movies</h3>
        </section>
        <section className='top-rated-movies'>
          {topMovies}
        </section>
        <section className='movies-heading'>
          <h3>All Movies</h3>
        </section>
        <section className='bottom-display-section'>
          {allMovies}
        </section>
    </section>
  ))
  }
}

const mapStateToProps = (state) => ({
  movies: state.movies,
  selectedMovie: state.selectedMovieReducer,
  user: state.user,
  ratings: state.ratings
})

const mapDispatchToProps = (dispatch) =>({
  addMoviesToStore: movies => dispatch(getMovies(movies)),
  addSelectedMovieToStore: movie => dispatch(updateSelectedMovie(movie))
})

export default connect(mapStateToProps, mapDispatchToProps)(MoviesContainer);
