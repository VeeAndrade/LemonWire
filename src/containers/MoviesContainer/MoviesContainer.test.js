import React from 'react';
import { MoviesContainer, mapStateToProps, mapDispatchToProps } from './MoviesContainer.js';
import { shallow } from 'enzyme';
import { getMoviesData } from '../../apiCalls';
import { updateSelectedMovie, getMovies } from '../../actions/index.js';

jest.mock('../../apiCalls')

describe('MoviesContainer', () => {

  beforeEach(() => {
    getMoviesData.mockImplementation(() => {
      return Promise.resolve([{
        id: 29,
        title: "Ford v Ferrari",
        poster_path: "https://image.tmdb.org/t/p/original//6ApDtO7xaWAfPqfi2IARXIzj8QS.jpg",
        backdrop_path: "https://image.tmdb.org/t/p/original//n3UanIvmnBlH531pykuzNs4LbH6.jpg",
        release_date: "2019-11-13",
        overview: "American car designer",
        average_rating: 1,
        user_rating: 0
      }]);
    });
  });

  it('should match the snapshot', () => {
    const user = {
      email: "sam@turing.io",
      id: 20,
      name: "Sam"
    }
    const wrapper = shallow(<MoviesContainer user={user} movies={[{
        id: 29,
        title: "Ford v Ferrari",
        poster_path: "https://image.tmdb.org/t/p/original//6ApDtO7xaWAfPqfi2IARXIzj8QS.jpg",
        backdrop_path: "https://image.tmdb.org/t/p/original//n3UanIvmnBlH531pykuzNs4LbH6.jpg",
        release_date: "2019-11-13",
        overview: "American car designer",
        average_rating: 1,
        user_rating: 0
      }]}/>)
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an array of movie objects', () => {
      const mockMoviesArray = [{
        id: 29,
        title: "Ford v Ferrari",
        poster_path: "image.jpg",
        backdrop_path: "image.jpg",
        release_date: "2019-11-13",
        overview: "American car designer",
        average_rating: 1,
        user_rating: 0
      }]
      const mockState = {
        movies: mockMoviesArray
      }
      const expected = {
        movies: mockMoviesArray}
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with updateSelectedMovie', () => {
      const mockDispatch = jest.fn();
      const selectedMovie = {
        id: 29,
        title: "Ford v Ferrari",
        poster_path: "image.jpg",
        backdrop_path: "image.jpg",
        release_date: "2019-11-13",
        overview: "American car designer",
        average_rating: 1,
        user_rating: 0
      }
      const actionToDispatch = updateSelectedMovie(selectedMovie);
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addSelectedMovieToStore(selectedMovie)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch with getMovies', () => {
      const mockDispatch = jest.fn();
      const moviesArr = [{
        id: 29,
        title: "Ford v Ferrari",
        poster_path: "image.jpg",
        backdrop_path: "image.jpg",
        release_date: "2019-11-13",
        overview: "American car designer",
        average_rating: 1,
        user_rating: 0
      }]
      const actionToDispatch = getMovies(moviesArr)
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addMoviesToStore(moviesArr)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  });
});