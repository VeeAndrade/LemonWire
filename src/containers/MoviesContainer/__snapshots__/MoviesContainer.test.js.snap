// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer MoviesContainer container/component should match the snapshot 1`] = `
<section
  className="movies-display-container"
>
  <section
    className="movies-heading"
  >
    <h3>
      Highy Rated Movies
    </h3>
  </section>
  <section
    className="top-rated-movies"
  >
    <MoviePreview
      key="30"
      movie={
        Object {
          "id": 30,
          "title": "Frozen II",
          "user_rating": 0,
        }
      }
      saveSelectedMovieToStore={[Function]}
      userLoggedIn={true}
    />
  </section>
  <section
    className="movies-heading"
  >
    <h3>
      All Movies
    </h3>
  </section>
  <section
    className="bottom-display-section"
  >
    <MoviePreview
      key="30"
      movie={
        Object {
          "id": 30,
          "title": "Frozen II",
          "user_rating": 0,
        }
      }
      saveSelectedMovieToStore={[Function]}
      userLoggedIn={true}
    />
  </section>
</section>
`;
